utils/modCommon.js

```diff
@@ -416,6 +418,34 @@ const modCommon = {
     await interaction.editReply({ content: `${target}'s nickname changed from ${u.escapeText(oldNick)} to ${u.escapeText(newNick)}.` });
   },
 
+  spamCleanup: async function(target, guild, auto = false) {
+    const unique = (items) => [...new Set(items)];
+    let toDelete = new u.Collection();
+    let deleted = 0;
+    let notDeleted = false;
+    const contents = auto ? unique(target.messages.map(m => m.content.toLowerCase())) : [target.content.toLowerCase()];
+    for (const [, channel] of guild.channels.cache) {
+      if (channel.isTextBased() && channel.messages.cache.size > 0) {
+        const messages = channel.messages.cache.filter(m => Date.now() - (config.spamThreshold.cleanupLimit * (auto ? 1 : 2)) <= m.createdTimestamp && m.author.id == (target.author?.id ?? target.id) && m.content.toLowerCase().includes(contents));
```

Comment:
check the message I sent you on messenger on fixing this. you need to compare the date of the first flagged message, not the current time.